crypto/cipher/nacl/decrypter.go

```diff
@@ -24,19 +23,14 @@ type Decrypter struct {
 
 // Decrypt data using recipient private key with AES in CBC mode.
 func (d Decrypter) Decrypt(data cipher.EncryptedContent) (cipher.PlainContent, error) {
-	privKeyBytes, err := validatePrivateKeyType(d.privateKey)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	return easyOpen(data, privKeyBytes)
+	return easyOpen(data, d.privateKey.Bytes()[32:])
 }
 
-func validatePrivateKeyType(pk crypto.PrivateKey) ([]byte, error) {
-	switch pk := pk.(type) {
+func validatePrivateKeyType(pk crypto.PrivateKey) error {
```

Comment:
I your instinct to trying and simplify this.  Let me give you a bit more information on why this method returns the `[]byte` because its not clear from the code.  `pk.Bytes()` returns the private key bytes, however this method has multiple uses (which is a bad thing!) its used for storing the private key in the keystore, decrypting and signing private keys. This needs to be resolved in the long term and I'll create an issue around that but there is a bit of investigation needed, I''ll raise it and would be great to get your comments.  This method then is responsible for ensuring the key is of a valid type but also extracting the bytes that are relevant for decryption. If `[32:]` is moved when calling `easyOpen` you have then it is not aware of what type of key it is and it may pick the incorrect bytes.

Reply:
I see. Thanks for the clarification